src/main/java/com/salesforce/dataloader/dao/database/DatabaseConfig.java

```diff
@@ -26,8 +26,8 @@
 package com.salesforce.dataloader.dao.database;
 
 import org.apache.commons.dbcp.BasicDataSource;
-import org.springframework.beans.factory.xml.XmlBeanFactory;
-import org.springframework.core.io.FileSystemResource;
+import org.springframework.context.ApplicationContext;
```

Comment:
Enables `PropertyPlaceholderConfigurer` to be used in **database-conf.xml**

Then the data source can load variables like `${db.username}` from a config file?

```
<!DOCTYPE beans PUBLIC "-//SPRING//DTD BEAN//EN" "http://www.springframework.org/dtd/spring-beans.dtd">
<beans>
    <bean id="propertyPlaceholderConfigurer" class="org.springframework.beans.factory.config.PropertyPlaceholderConfigurer">
        <property name="locations">
            <list>
                <value>classpath:database.properties</value>
            </list>
        </property>
    </bean>
    <bean class="org.apache.commons.dbcp.BasicDataSource" destroy-method="close" id="MsqlDataSource">
        <property name="username" value="${db.user}"/>
        ...
```

http://forum.spring.io/forum/spring-projects/container/58380-propertyplaceholderconfigurer-is-ignored